### ISSUE TITLE
Week 7 projects being an extension of week 6?

### ISSUE THREAD

**Issue by astroash** [foundersandcoders/master-reference#764]
Week 7 projects involve students implementing `TDD`, `a vanilla node server`, `database`, `database testing` and `authentication in a 2 day project`. In Nazareth they made week 7's project an extension of week 6's project and it went really. In London they said they [wished this happened](https://github.com/foundersandcoders/london-programme/blob/master/stop-go-continue/fac-11/cohort-week-7.md#was-project-too-big-bart) as it was too much to fit in. I propose that we implement Nazareth's approach and make week 7's projects an extension of week 6's.

For this to be implemented:
- Update both week project files to reflect this
- Week 6 mentors will have to mention this when introducing topics
- Team switching will have to not happen between these two weeks

**Comment by shiryz** [comment]
@astroash do you think we should split the 4s into pairs since the project is not built from scratch?

**Comment by astroash** [comment]
@shiryz how do you mean?

I think my preferred option would be the team of 4 staying together and continuing on the same project.

**Comment by ZooeyMiller** [comment]
As long as it is clear in week 6 that this project will be continued for week 7, then I don't see this as a problem. I didn't see week 7 in any location so I don't know how it went. I will also not be in Gaza when this happens for this cohort so I think the opinion of @mantagen would be valuable here too.

**Comment by mattlub** [comment]
for the record, in N2:
- they were not told anything extra in week 6
- they were given the choice in week 7 to continue or not (all did)
- the teams had changed from w6 to w7 (so 2 teams actually ended up building on the same project)

**Comment by m4v15** [comment]
Yeah I think this is a good idea but not sure it _has_ to mentioned in week 6. I can see how it might seem necessary, but actually, what is the benefit of telling people this? I feel like in some situations that actually could confuse things maybe? Just because it happened in FACN1 and FACN2 without it being telegraphed the week before and it was fine, but obviously different circumstances and situations might change this

**Comment by astroash** [comment]
Makes sense, I'm convinced it doesn't need to be mentioned in week 6. Curious to hear peoples opinions on whether groups should be kept the same, it's interesting that @mattlub says this didn't happen in Nazareth, were there any issues with this?

**Comment by ZooeyMiller** [comment]
@astroash I think it makes sense for groups to be the same. Having the prewritten (by someone else) code in week 2 is fine because we know it's of good quality. We can't guarantee people's code from week 6 will be of a reasonable quality, if this is your own code, that you carry on working on, then this is fine, but if you have to pick up another groups' spaghetti code, then it may be just as time consuming to decipher it, as it would have been just to have it as a new project. Not to mention, probably quite frustrating. 

@m4v15 In my eyes, the only reason we wouldn't tell the students they'll be working on the same project next week, would be if we didn't know at that point. I don't see how it could be more confusing to tell the students this, than to not tell them? 

I also think, considering that the project for week 6 is quite free-form (basically just build an app with a database), knowing that the following week they are going to implement Auth on it could help guide what they decide to build.

**Comment by shiryz** [comment]
@astroash they did build on the projects from week 6 but all were given the choice to build something from scratch.
I don't think changing the teams was so terrible, they had to familiarise themselves with someone else's code and understand what's going on, but agreed this is not the best use of their time in a 2-day project, my original suggestion was splitting the team in half so 8, 2 people teams.
@ZooeyMiller not telling them that it's the same project led to having a new plan built on the same code, even though it was built on the same idea.

**Comment by ZooeyMiller** [comment]
I'm happy to go with the views of the people who have seen this in practise, so if the Naz mentors who oversaw/had this process in their cohort think it will work, then I'm comfortable with the process for the current cohorts being as discussed. I just wanted to voice my concerns.

**Comment by astroash** [comment]
@ZooeyMiller I totally get your concerns. I would also be worried about diving into unknown code of a variable quality in a short amount of time. How about CFs in each campus pick the variations in implementing this which suit their cohort, and we can discuss afterwards whether we want set rules in mazref about it?

We're on week 5 in London at the moment on a 2 week group rotation, so I'm implementing no team swaps between 6-7 I need to decide by lunchtime today. Instead of doing two week swaps I'll move to 2 weeks, 3 weeks, 3 weeks.

**Comment by m4v15** [comment]
@astroash yeah I think actually each CF implementing this as they see fit is the right way to go on this, I don't necessarily think just cos what worked in Naz before will work everywhere else.

@ZooeyMiller yeah I guess I could just imagine people then worrying in week 6 about auth in week 7, but actually I'm probably overstating this and in fact maybe it could help like you say

**Comment by ZooeyMiller** [comment]
I personally still don't really like the idea of it for Gaza right now, but I won't be CF when this is happening, @mantagen will be, so his opinion is more important than mine, really.
